From 4061fc04dd3ef24609eeedd875085d71eed4a3cf Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Sat, 3 Jan 2026 14:18:26 +0800
Subject: [PATCH 55/63] loonggpu-bridge: make bridge_phy_alloc() return
 ERR_PTR(-ENOMEM) instead of NULL on alloc failure

Some logics for 7a2000 and 2k3000 are checking the return value with
IS_ERR.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu-bridge/bridge_phy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/loonggpu-bridge/bridge_phy.c b/loonggpu-bridge/bridge_phy.c
index e0e5eb5..487b5bf 100644
--- a/loonggpu-bridge/bridge_phy.c
+++ b/loonggpu-bridge/bridge_phy.c
@@ -773,7 +773,7 @@ struct loonggpu_bridge_phy *bridge_phy_alloc(struct loonggpu_dc_bridge *dc_bridg
 	bridge_phy = devm_kzalloc(dc_bridge->adev->dev, sizeof(*bridge_phy), GFP_KERNEL);
 	if (!bridge_phy) {
 		DRM_ERROR("Failed to alloc loonggpu bridge phy!\n");
-		return bridge_phy;
+		return ERR_PTR(-ENOMEM);
 	}
 #endif
 
-- 
2.52.0

